src/main/java/universe/UniverseViewpointAdapter.java

```diff
@@ -32,28 +25,37 @@ protected AnnotationMirror combineAnnotationWithAnnotation(
         assert receiverAnnotation != null;
         assert declaredAnnotation != null;
 
-        if (AnnotationUtils.areSame(receiverAnnotation, SELF)) {
+        if (AnnotationUtils.areSame(
+                receiverAnnotation, ((UniverseAnnotatedTypeFactory) atypeFactory).SELF)) {
```

Comment:
Same comment about extracting the casts by adding a field or even just a local variable here.